functions.sh

```diff
@@ -239,3 +239,54 @@ function sort_versions() {
 
   echo "${sorted[@]}"
 }
+
+function commit_range() {
+  local commit_id_end=${1}
+  shift
+  local commit_id_start=${1}
+
+  if [ -z "${commit_id_start}" ] && [ -z "${commit_id_end}" ]; then
+    echo "HEAD~1..HEAD"
+  elif [ -z "${commit_id_start}" ] && [[ "${commit_id_end}" =~ .. ]]; then
+    echo "${commit_id_end}"
+  elif [ -z "${commit_id_start}" ]; then
```

Comment:
Consider to combine  `[ -z "${commit_id_start}" ]` condition?

Reply:
All of these conditions are required and produce unique output

If you don't send anything it uses HEAD~1..HEAD

If you provide just the first one and there is a .. in it then it's a range so output as is

If you provide just the first one and there is no .. then it's a single commit so ouput commit_id~1..commit_id

Finally, if you provide both then it's
commit_id2..commid_id1

Reply:
I mean to put them all under a ` [ -z "${commit_id_start}" ]` condition, there is no need to write the same condition in 3 places, right?

Reply:
Aaah yes! Good point!!

Reply:
Done!